WordVectors/python/core_notebook.ipynb

```diff
@@ -0,0 +1,786 @@
+    "\n",
+    "We are going to start by writing a function that will perform our cleaning tasks. This way, if we want to clean other data later on, it is easy to pass that data into this function.\n",
+    "\n",
+    "### The Code ###\n",
+    "\n",
+    "In the code block below, we define a function called `clean_text` which accepts a list of texts, called `text` in the definition, as a parameter. \n",
+    "\n",
+    "The first thing we do is to split our list of texts into a list of individual words. Word embedding models work with individual words, so we use the `.split()` function to take a list that may look like this: \n",
+    "\n",
+    "`[\"this is a string\"]`\n",
+    "\n",
+    "into a list that looks like this:\n",
+    "\n",
+    "`[\"this\", \"is\", \"a\", \"string\"]`\n",
+    "\n",
+    "The process of splitting a text into its individual words is called **tokenizing** and the individual words are called **tokens**. We will use the variabel `tokens` to hold our tokenized data.\n",
+    "\n",
+    "Then, we're going to make all of the words in our `tokens` list lowercase. We accomplish this using the built-in `.lower()` function and what is called **list comprehension**. List comprehension allows us to create a new list using what the computer understands about our old list. In the code block below, we run: \n",
+    "\n",
+    "    t.lower() for t in tokens\n",
+    "    \n",
+    "This abbreviated for loop uses list comprehension to iterate through each item in `tokens` and apply the `.lower()` function to each of those individual items. We store the resulting list in the variable `tokens` which will overwrite the old `tokens` list.\n",
+    "\n",
+    "Then, using the `re` library which gives us access to regular expressions, we use list comprehension to iterate through each of the items in our new `tokens` list and substitute all instances of punctuation with an empty string, `''`, this effectively removes the punctuation. We store this edited list in the `tokens` variable which will overwrite the old `tokens` variable just like before. \n",
+    "\n",
+    "Finally, we remove any items from our list that are not alphabetical characters (for instance, numbers and special characters) using list comprehension and the `isalpha()` function. We overwrite the `tokens` variable one last time with our final verison of the tokenized data and then return `tokens` which will give the rest of the code access to the tokenized text."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def clean_text(text):\n",
+    "    \n",
+    "    # Cleans the given text using regular expressions to split and lower-cased versions to create\n",
+    "    # a list of tokens for each text.\n",
+    "    # The function accepts a list of texts and returns a list of of lists of tokens\n",
```

Comment:
The function is supposed to accept a list where each item coorresponds to the text of a single file (so it would look like [text1, text2, text3]. where text1, for example, is the all of the text in that file represented as a string (that is what the data variable is supposed to be) It really accepts any list of strings, so you could also have a list where each item is a sentence from the same text or any other format. Is the data variable not in that format for you and if it is, I'll try to think of a more clear way to explain the parameters

Reply:
`data` does have a list of texts, but the code block isn’t providing it directly to the function:

```py
# clean text from folder of text files, stored in the data variable
data_clean = []
for x in data:
    data_clean.append(clean_text(x))
```

If `clean_text()` accepted the list as input I would expect this to work:

```py
clean_text(data)
```

Reply:
(To be clear, I think it’s fine that the function doesn’t include the `for` loop. The description of the code just needs to be updated for accuracy.)